File: 1598-crawler-log-folder/main.go
func minOperations(logs []string) int {
    
    var depth int = 0

    for _, log := range logs {
        switch log {
        case "../":
            if depth > 0 {
                depth -= 1
            }
        case "./":
            continue
        default:
            depth += 1
        }
    }

    return depth
}

File: 1598-crawler-log-folder/main.rs
impl Solution {
    pub fn min_operations(logs: Vec<String>) -> i32 {
        let mut depth = 0;

        for log in logs.iter() {
            match log.as_str() {
                "../" => {
                    if depth > 0 {
                        depth -= 1;
                    }
                },
                "./" => continue,
                _ => depth += 1,
            }
        }

        depth
    }
}
